### PR TITLE
Bump OpenTelemetry sdk to 1.42.1

### PR DESCRIPTION
**Pull request by brunobat**
- Instrumentation to 2.8.0 
- Semantic conventions to 1.27.0
- Point deprecated conventions to the new classes.

### CODE REVIEW

**Comment by brunobat** [comment]
Do not merge for now, because of possible side effects on Microprofile Telemetry 2.0 certification. Set as draft.

**Comment by brunobat** [comment]
Needed for https://github.com/quarkusio/quarkus/pull/43678

File: extensions/opentelemetry/runtime/pom.xml [hunk 1]
```diff
@@ -98,6 +98,10 @@
             <groupId>io.opentelemetry.semconv</groupId>
             <artifactId>opentelemetry-semconv</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.opentelemetry.semconv</groupId>
+            <artifactId>opentelemetry-semconv-incubating</artifactId>
```
[the remark this answers is not in the document]
Comment (brunobat) [reply extensions/opentelemetry/runtime/pom.xml:103]: Yes they did. The idea is that all unstable conventions will be placed there and later moved to the main artifact once final.
Unfortunately there is still a lot of unstable content.

**Review by radcortez — APPROVED** [review]

**Comment by zakkak** [comment]
@brunobat after this change we see a `SEVERE` error instead of a `WARNING` we used to get in https://github.com/Karm/mandrel-integration-tests (note that we don't really want to setup and test any OpenTracing collector).

See https://github.com/Karm/mandrel-integration-tests/issues/266#issuecomment-2410390365

## Before:

```
2024-10-14 12:34:11,656 WARNING [io.ope.exp.int.grp.GrpcExporter] (vert.x-eventloop-thread-1) Failed to export spans. Server responded with gRPC status code 2. Error message: Failed to export TraceRequestMarshalers. The request could not be executed. Full error message: Connection refused: localhost/127.0.0.1:4317
```

## After

```
2024-10-14 12:31:28,716 SEVERE [io.ope.exp.int.grp.GrpcExporter] (vert.x-eventloop-thread-1) Failed to export spans. The request could not be executed. Error message: Connection refused: localhost/127.0.0.1:4317: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:4317
	Suppressed: java.lang.IllegalStateException: Retries exhausted: 3/3
		at io.smallrye.mutiny.helpers.ExponentialBackoff$1.lambda$apply$0(ExponentialBackoff.java:46)
		at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
		at io.smallrye.mutiny.groups.MultiOnItem.lambda$transformToUni$6(MultiOnItem.java:268)
		at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber.onItem(MultiConcatMapOp.java:117)
		at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
		at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drainWithDownstream(UnicastProcessor.java:108)
		at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drain(UnicastProcessor.java:139)
		at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.onNext(UnicastProcessor.java:205)
		at io.smallrye.mutiny.operators.multi.processors.SerializedProcessor.onNext(SerializedProcessor.java:104)
		at io.smallrye.mutiny.subscription.SerializedSubscriber.onItem(SerializedSubscriber.java:74)
		at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
		at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.onFailure(MultiRetryWhenOp.java:127)
		at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
		at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:104)
		at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap$UniOnFailureFlatMapProcessor.dispatch(UniOnFailureFlatMap.java:85)
		at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap$UniOnFailureFlatMapProcessor.onFailure(UniOnFailureFlatMap.java:60)
		at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:60)
		at java.base@21.0.4/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
		at java.base@21.0.4/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
		at java.base@21.0.4/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base@21.0.4/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
		at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:603)
		at io.vertx.core.impl.future.FutureImpl$4.onFailure(FutureImpl.java:188)
		at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
		at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
		at io.vertx.core.impl.future.Mapping.onFailure(Mapping.java:45)
		at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
		at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
		at io.vertx.core.http.impl.HttpClientImpl.lambda$doRequest$4(HttpClientImpl.java:398)
		at io.vertx.core.net.impl.pool.Endpoint.lambda$getConnection$0(Endpoint.java:52)
		at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:162)
		at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:123)
		at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:328)
		at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:321)
		at io.vertx.core.net.impl.pool.SimpleConnectionPool$ConnectFailed$1.run(SimpleConnectionPool.java:380)
		at io.vertx.core.net.impl.pool.Task.runNextTasks(Task.java:43)
		at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:91)
		at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:244)
		at io.vertx.core.net.impl.pool.SimpleConnectionPool.lambda$connect$2(SimpleConnectionPool.java:258)
		at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint.lambda$connect$2(SharedClientHttpStreamEndpoint.java:104)
		at io.vertx.core.impl.future.FutureImpl$4.onFailure(FutureImpl.java:188)
		at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
		at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
		at io.vertx.core.impl.future.Composition$1.onFailure(Composition.java:66)
		at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
		at io.vertx.core.impl.future.FailedFuture.addListener(FailedFuture.java:98)
		at io.vertx.core.impl.future.Composition.onFailure(Composition.java:55)
		at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
		at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
		at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:328)
		at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:321)
		at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:352)
		at io.vertx.core.net.impl.NetClientImpl.lambda$connectInternal2$6(NetClientImpl.java:324)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
		at io.netty.util.concurrent.DefaultPromise.setFailure(DefaultPromise.java:110)
		at io.vertx.core.net.impl.ChannelProvider.lambda$handleConnect$0(ChannelProvider.java:157)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118)
		at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:326)
		at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:342)
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base@21.0.4/java.lang.Thread.runWith(Thread.java:1596)
		at java.base@21.0.4/java.lang.Thread.run(Thread.java:1583)
		at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
		at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
	Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:4317
		Suppressed: [CIRCULAR REFERENCE: java.lang.IllegalStateException: Retries exhausted: 3/3]
	Caused by: java.net.ConnectException: Connection refused
		at java.base@21.0.4/sun.nio.ch.Net.pollConnect(Native Method)
		at java.base@21.0.4/sun.nio.ch.Net.pollConnectNow(Net.java:682)
		at java.base@21.0.4/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
		at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336)
		at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339)
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base@21.0.4/java.lang.Thread.runWith(Thread.java:1596)
		at java.base@21.0.4/java.lang.Thread.run(Thread.java:1583)
		at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
		at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
Caused by: [CIRCULAR REFERENCE: java.net.ConnectException: Connection refused]

```

Is this the expected behaviour? 

cc @Karm

**Comment by brunobat** [comment]
@zakkak that test seems to be the one related with the OpenTelemetry collector... You don't need to setup the collector, the test should do it out of the box with Docker, but for some reason, the sender request is not reaching the collector: `...Connection refused: localhost/127.0.0.1:4317...`

**Comment by zakkak** [comment]
That's intended AFAIK (@Karm please confirm). The question is, should the application report a SEVERE error in such cases? Till now it used to just print a WARNING.

**Comment by brunobat** [comment]
I'm afraid this comes from the upstream commit: https://github.com/open-telemetry/opentelemetry-java/commit/1f6de3590d284e3ced20c5ac9c8217dda6336732

**Pull request merged**